tools/rimage/src/module.c

```diff
@@ -252,7 +249,7 @@ static void sections_info_add(struct module_sections_info *info, struct module_s
  */
 static void sections_info_finalize(struct module_sections_info *info)
 {
-	info->file_size = info->end - info->start;
+	info->file_size = (info->end > info->start) ? info->end - info->start : 0;
```

Comment:
sorry, don't understand. You say, previously this produced the result of 1, i.e. `info->end - info->start == 1` which means `info->end > info->start` so now this still would calculate 1. What am I missing?

Reply:
The initial state is
`info->start = UINT32_MAX;
info->end = 0;`

Reply:
is there a particular reason for that? Could you just initialise both to 0?

Reply:
The segment size increases as more sections are added from the input file. When adding, the `start` and `end` markers are moved. The `start` decreases if the section starts earlier than the current start. The `end` increases if the end of the section extends beyond the current segment end. This is a classic min(start) and max(end) algorithms, hence the initial values.

Reply:
@lyakh The `size` variable is initialized to 0. The `start` marker is initialized to the maximum value to simplify calculation of the minimum starting address, and to distinguish a valid value of 0 from the initial state.